issue of installing&Changing libraries

Hi,
I successfully installed python-fsps. And want to use PADOVA as the isochron library MILES as the stellar spectrum library, and follow the instructions to uninstall and change the library and reinstall. However, the installation was not successful. 
Could you provide some insights into this?Thanks for your guidance!
The installation process and error information are as follows:

```shell
(pixedfit2) kongxiangming@agn:~/Downloads/fsps-3.2/src$ python -m pip install fsps
Collecting fsps
  Downloading fsps-0.4.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.4 kB)
Collecting numpy (from fsps)
  Using cached numpy-1.26.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)
Downloading fsps-0.4.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 944.9 kB/s eta 0:00:00
Using cached numpy-1.26.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
Installing collected packages: numpy, fsps
Successfully installed fsps-0.4.5 numpy-1.26.1
(pixedfit2) kongxiangming@agn:~/Downloads/fsps-3.2/src$ pip uninstall fsps
Found existing installation: fsps 0.4.5
Uninstalling fsps-0.4.5:
  Would remove:
    /home/kongxiangming/.conda/envs/pixedfit2/lib/python3.10/site-packages/fsps-0.4.5.dist-info/*
    /home/kongxiangming/.conda/envs/pixedfit2/lib/python3.10/site-packages/fsps.libs/libgfortran-040039e1.so.5.0.0
    /home/kongxiangming/.conda/envs/pixedfit2/lib/python3.10/site-packages/fsps.libs/libquadmath-96973f99.so.0.0.0
    /home/kongxiangming/.conda/envs/pixedfit2/lib/python3.10/site-packages/fsps/*
Proceed (Y/n)? y
  Successfully uninstalled fsps-0.4.5
(pixedfit2) kongxiangming@agn:~/Downloads/fsps-3.2/src$ FFLAGS="-DMIST=0 -DPADOVA=1 -DMILES=1 -DBASEL=0" python -m pip install fsps --no-binary fsps
Collecting fsps
  Using cached fsps-0.4.5.tar.gz (54 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy in /home/kongxiangming/.conda/envs/pixedfit2/lib/python3.10/site-packages (from fsps) (1.26.1)
Building wheels for collected packages: fsps
  Building wheel for fsps (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for fsps (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [36 lines of output]
      *** scikit-build-core 0.5.1 using CMake 3.27.7 (wheel)
      *** Configuring CMake...
      2023-10-20 17:11:05,408 - scikit_build_core - WARNING - libdir/ldlibrary: /home/kongxiangming/.conda/envs/pixedfit2/lib/libpython3.10.a is not a real file!
      2023-10-20 17:11:05,408 - scikit_build_core - WARNING - Can't find a Python library, got libdir=/home/kongxiangming/.conda/envs/pixedfit2/lib, ldlibrary=libpython3.10.a, multiarch=x86_64-linux-gnu, masd=None
      loading initial cache file /tmp/tmpl_bb60du/build/CMakeInit.txt
      -- The C compiler identification is GNU 7.5.0
      -- The Fortran compiler identification is GNU 7.5.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/gcc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting Fortran compiler ABI info
      -- Detecting Fortran compiler ABI info - done
      -- Check for working Fortran compiler: /usr/bin/gfortran - skipped
      -- Found Python: /home/kongxiangming/.conda/envs/pixedfit2/bin/python (found version "3.10.13") found components: Interpreter Development.Module NumPy
      -- Configuring done (1.3s)
      CMake Error at /tmp/pip-build-env-rwru3336/normal/lib/python3.10/site-packages/cmake/data/share/cmake-3.27/Modules/FindPython/Support.cmake:4035 (add_library):
        Cannot find source file:
      
          /tmp/pip-install-puzkp0w9/fsps_8534c0ebc53f480ebd10627ff2cf21d0/src/fsps/libfsps/src/sps_vars.f90
      Call Stack (most recent call first):
        /tmp/pip-build-env-rwru3336/normal/lib/python3.10/site-packages/cmake/data/share/cmake-3.27/Modules/FindPython.cmake:635 (__Python_add_library)
        src/fsps/CMakeLists.txt:64 (python_add_library)
      
      
      CMake Error at /tmp/pip-build-env-rwru3336/normal/lib/python3.10/site-packages/cmake/data/share/cmake-3.27/Modules/FindPython/Support.cmake:4035 (add_library):
        No SOURCES given to target: _fsps
      Call Stack (most recent call first):
        /tmp/pip-build-env-rwru3336/normal/lib/python3.10/site-packages/cmake/data/share/cmake-3.27/Modules/FindPython.cmake:635 (__Python_add_library)
        src/fsps/CMakeLists.txt:64 (python_add_library)
      
      
      CMake Generate step failed.  Build files cannot be regenerated correctly.
      
      *** CMake configuration failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for fsps
Failed to build fsps
ERROR: Could not build wheels for fsps, which is required to install pyproject.toml-based projects
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -48,6 +48,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
+          submodules: true
           fetch-depth: 0
       - uses: actions/setup-python@v4
         with:

--- a/src/fsps/CMakeLists.txt
+++ b/src/fsps/CMakeLists.txt
@@ -1,74 +1,79 @@
+if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_vars.f90")
+  message(FATAL_ERROR "The source code in the FSPS submodule was not found."
+    "Please run 'git submodule update --init' to initialize the submodule.")
+endif()
+
 # Generate the f2py wrappers
 set(F2PY_SOURCES
-    "${CMAKE_CURRENT_BINARY_DIR}/_fspsmodule.c"
-    "${CMAKE_CURRENT_BINARY_DIR}/_fsps-f2pywrappers2.f90"
+  "${CMAKE_CURRENT_BINARY_DIR}/_fspsmodule.c"
+  "${CMAKE_CURRENT_BINARY_DIR}/_fsps-f2pywrappers2.f90"
 )
 add_custom_command(
   OUTPUT ${F2PY_SOURCES}
   DEPENDS fsps.f90
   VERBATIM
   COMMAND "${Python_EXECUTABLE}" -m numpy.f2py
-          "${CMAKE_CURRENT_SOURCE_DIR}/fsps.f90" -m _fsps --lower
-          --build-dir "${CMAKE_CURRENT_BINARY_DIR}")
+  "${CMAKE_CURRENT_SOURCE_DIR}/fsps.f90" -m _fsps --lower
+  --build-dir "${CMAKE_CURRENT_BINARY_DIR}")
 
 # List out the explicit FSPS sources; we don't use a glob here since
 # the 'src' directory includes some executables that we don't want to
 # include here.
 set(
-    FSPS_SOURCES
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_vars.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_utils.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_agb_dust.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_bs.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_dust.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_nebular.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_remnants.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_xrb.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/agn_dust.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/attn_curve.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/compsp.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/csp_gen.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/funcint.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/get_lumdist.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/get_tuniv.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/getindx.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/getmags.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/getspec.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/igm_absorb.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/imf.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/imf_weight.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/intsfwght.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/linterp.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/linterparr.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/locate.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/mod_gb.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/mod_hb.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/pz_convol.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sbf.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/setup_tabular_sfh.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfh_weight.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfhinfo.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfhlimit.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfhstat.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/smoothspec.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/spec_bin.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_setup.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/ssp_gen.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/tsum.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/vacairconv.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/write_isochrone.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/ztinterp.f90"
+  FSPS_SOURCES
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_vars.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_utils.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_agb_dust.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_bs.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_dust.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_nebular.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_remnants.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/add_xrb.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/agn_dust.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/attn_curve.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/compsp.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/csp_gen.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/funcint.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/get_lumdist.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/get_tuniv.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/getindx.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/getmags.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/getspec.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/igm_absorb.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/imf.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/imf_weight.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/intsfwght.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/linterp.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/linterparr.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/locate.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/mod_gb.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/mod_hb.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/pz_convol.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sbf.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/setup_tabular_sfh.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfh_weight.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfhinfo.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfhlimit.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sfhstat.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/smoothspec.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/spec_bin.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/sps_setup.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/ssp_gen.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/tsum.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/vacairconv.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/write_isochrone.f90"
+  "${CMAKE_CURRENT_SOURCE_DIR}/libfsps/src/ztinterp.f90"
 )
 
 # Define the Python library
 python_add_library(
-    _fsps
-    MODULE
-    "${CMAKE_CURRENT_SOURCE_DIR}/fsps.f90"
-    ${FSPS_SOURCES}
-    ${F2PY_SOURCES}
-    "${F2PY_INCLUDE_DIR}/fortranobject.c"
-    WITH_SOABI)
+  _fsps
+  MODULE
+  "${CMAKE_CURRENT_SOURCE_DIR}/fsps.f90"
+  ${FSPS_SOURCES}
+  ${F2PY_SOURCES}
+  "${F2PY_INCLUDE_DIR}/fortranobject.c"
+  WITH_SOABI)
 target_link_libraries(_fsps PUBLIC Python::NumPy)
 target_include_directories(_fsps PUBLIC "${F2PY_INCLUDE_DIR}")
 install(TARGETS _fsps DESTINATION ${SKBUILD_PROJECT_NAME})

